SYS_ATL/LoopIR_scheduling.py

```diff
@@ -1799,6 +1799,127 @@ def map_s(self, s):
+                                eff     = self.orig_proc.eff,
+                                srcinfo = self.orig_proc.srcinfo)
+        self.proc = InferEffects(self.proc).result()
+
+    def result(self):
+        return self.proc
+
+    def alloc_check(self, pre, post):
+        if not _is_alloc_free(pre, post):
+            raise SchedulingError("Will not fission here, because "
+                                  "an allocation might be buried "
+                                  "in a different scope than some use-site")
+
+    # returns a pair of stmt-lists
+    # for those statements occurring before and
+    # after the fission point
+    def map_stmts(self, stmts):
+        pre_stmts           = []
+        post_stmts          = []
+        for orig_s in stmts:
+            pre, post       = self.map_s(orig_s)
+            pre_stmts      += pre
+            post_stmts     += post
+
+        return (pre_stmts, post_stmts)
+
+    # see map_stmts comment
+    def map_s(self, s):
+        if s is self.tgt_stmt:
+            #assert self.hit_fission == False
+            self.hit_fission = True
+            # none-the-less make sure we return this statement in
+            # the pre-fission position
+            return ([s],[])
+
+        elif isinstance(s, LoopIR.If):
+
+            # first, check if we need to split the body
+            pre, post = self.map_stmts(s.body)
+            fission_body = (len(pre) > 0 and len(post) > 0 and self.n_lifts > 0)
+            if fission_body:
+                self.n_lifts -= 1
+                self.alloc_check(pre, post)
+                pre = LoopIR.If(s.cond, pre, [], None, s.srcinfo)
+                post = LoopIR.If(s.cond, post, s.orelse, None, s.srcinfo)
+                return ([pre],[post])
+
+            body = pre+post
+
+            # if we don't, then check if we need to split the or-else
+            pre, post       = self.map_stmts(s.orelse)
+            fission_orelse  = (len(pre) > 0 and len(post) > 0 and
+                               self.n_lifts > 0)
+            if fission_orelse:
+                self.n_lifts -= 1
+                self.alloc_check(pre, post)
+                pre         = LoopIR.If(s.cond, body, pre, None, s.srcinfo)
+                post        = LoopIR.If(s.cond, [LoopIR.Pass(None, s.srcinfo)],
+                                                post, None, s.srcinfo)
+                return ([pre],[post])
+
+            orelse = pre+post
+
+            # if we neither split the body nor the or-else,
+            # then we need to gather together the pre and post.
+            single_stmt = LoopIR.If(s.cond, body, orelse, None, s.srcinfo)
+
+        elif isinstance(s, LoopIR.ForAll) or isinstance(s, LoopIR.Seq):
+
+            # check if we need to split the loop
+            pre, post = self.map_stmts(s.body)
+            do_fission = (len(pre) > 0 and len(post) > 0 and
+                          self.n_lifts > 0)
+            if do_fission:
```

Comment:
This and other checks like it can be replaced with `if pre and post and self.n_lifts > 0:`.

In general, don't use `len()` to check if a list is empty or not.